lib/matplotlib/projections/geo.py

```diff
@@ -269,24 +269,26 @@ def __init__(self, resolution):
             self._resolution = resolution
 
         def transform_non_affine(self, ll):
-            longitude = ll[:, 0:1]
-            latitude  = ll[:, 1:2]
+            longitude = ll[:, 0]
+            latitude = ll[:, 1]
 
             # Pre-compute some values
             half_long = longitude / 2.0
             cos_latitude = np.cos(latitude)
 
             alpha = np.arccos(cos_latitude * np.cos(half_long))
             # Mask this array or we'll get divide-by-zero errors
-            alpha = ma.masked_where(alpha == 0.0, alpha)
-            # The numerators also need to be masked so that masked
-            # division will be invoked.
+            mask = alpha != 0.0
+            alpha = alpha[mask]
             # We want unnormalized sinc.  numpy.sinc gives us normalized
-            sinc_alpha = ma.sin(alpha) / alpha
+            sinc_alpha = np.sin(alpha) / alpha
 
-            x = (cos_latitude * ma.sin(half_long)) / sinc_alpha
-            y = (ma.sin(latitude) / sinc_alpha)
-            return np.concatenate((x.filled(0), y.filled(0)), 1)
+            xy = np.empty_like(ll, float)
+            xy[mask, 0] = ((cos_latitude[mask] * np.sin(half_long[mask])) /
+                           sinc_alpha)
+            xy[mask, 1] = np.sin(latitude[mask]) / sinc_alpha
```

Comment:
Can sinc_alpha be zero?  If so, the original masked array version was handling that case and the present version does not appear to be doing so.

Reply:
`sinc_alpha` comes from `sin(alpha)`, which can only be zero if `alpha == nπ`. `alpha` comes from `arccos`, whose output is [0, π] and since 0 is masked, we only need to worry about π which is from input of -1, from `cos_latitude * np.cos(half_long)`. The limits are -π/2 ≤ latitude ≤ π/2 and -π ≤ longitude ≤ π (and cannot be changed), that's also -π/2 ≤ `half_long` ≤ π/2, so both `cos` terms (and the entire input) are [0, 1]. Thus it's impossible for `sinc_alpha` to be zero.

Reply:
Thank you.   (Sorry, I was too lazy to figure that out for myself.)